R/functions.R

```diff
@@ -522,25 +518,58 @@ df2vw <- function(data, file_path, namespaces = NULL, keep_space = NULL,
     # Construct vw format for labels, weights, base and tag
     formatDataVW <- ""
     argexpr <- c()
-    names_indices <- c(1:ncol(data))
+    names_indices <- seq_len(ncol(data))
     names(names_indices) <- names(data)
     
     if(!is.null(targets)) {
         formatDataVW <- paste0(formatDataVW, "%s")
         if(length(targets) > 1) {
+
+            # Initialize empty labels
+            data[["parsed_labels"]] <- rep("", nrow(data))
+            
             if(!is.null(probabilities)) {
+                
                 if(length(targets) != length(probabilities)) {
                     stop("targets and probabilities should be of the same length")
                 }
-                temp_labels <- paste0("eval(parse(text = 'constructLabelsCB(targets, probabilities, nrow(data))'))")
-                argexpr <- c(argexpr, temp_labels)
+                
+                # Construct Labels for multilabel examples with probabilities (e.g. 1:0.6:0.3 2:0.4:0.7)
+                # Iterate cost vectors names
+                for(i in 1:length(targets)) {
```

Comment:
Humor me here and switch from `1:length(targets)` to either: `seq_len(length(targets))` or `seq_along(targets)`. This provides a protected iteration index. 

e.g. if `length(targets)` is 0, then the loop will run as `1:0` becomes a descending sequence `c(1, 0)`.